### ISSUE TITLE
🐛 insertHeadingbreak Bug

### ISSUE THREAD

**Issue by d-e-v-esh** [accordproject/web-components#260]
# Bug Report 🐛

The cursor would jump one line above when we would press enter from the beginning of any heading text

![r91hU4x2ti](https://user-images.githubusercontent.com/59534570/108851223-36701e00-760a-11eb-8a6a-38c252ce60b6.gif)

## Expected Behavior

The cursor should go down with the heading text as it happens everywhere else

## Steps to Reproduce
1. Put the cursor to the beginning of any heading and press enter.

### Desktop
 - OS: Windows 10
 - Browser: Chrome
 - Version 88.0.4324.182 (Official Build) (64-bit)

**Comment by d-e-v-esh** [comment]
@irmerk This problem is occurring because:
When we press enter from the start of the heading word => it inserts a line above the heading line by default instead of below.
This only happens when we insert from the start of the line. It works perfectly when we do it from the middle or the end of the heading word.
The fix for this can be:
1. Whenever ``headingbreak`` is called => the next line node it inserts down below would be converted into a paragraph.
2. This creates a new problem where if we press enter from the beginning of the heading word, the heading would be converted into a paragraph too.
3. To fix that=> Override the default ``insertbreak()`` in the ``headingbreak`` function and tell it to do the above 1. only when it is at the end of the line. We create a ``point`` for the end of the heading line and make a conditional to check if the ``insetbreak()`` is being called from the stored end of the line point and execute 1 if that is true.

Is that an appropriate solution?

**Comment by irmerk** [comment]
If I understand correctly, I think that sounds good. I think we want:

- <kbd>ENTER</kbd> at BEGINNING of node (where ↑ is) creates a new line of the same text: 
```html
<h1>Heading</h1>
↑
```
```html
<p>Normal paragraph text.</p>
↑
```
Becomes this with the cursor (↑) positioned:
```html
<h1></h1>
<h1>Heading</h1>
↑
```
```html
<p></p>
<p>Normal paragraph text.</p>
↑
```
---
- <kbd>ENTER</kbd> at MIDDLE of node (where ↑ is) creates a new line of similar text, essentially splitting the node:
```html
<h1>Heading</h1>
       ↑
```
```html
<p>Normal paragraph text.</p>
       ↑
```
Becomes this with the cursor (↑) positioned:
```html
<h1>Hea</h1>
<h1>ding</h1>
↑
```
```html
<p>Norm</p>
<p>al paragraph text.</p>
↑
```
---
- <kbd>ENTER</kbd> at END of node (where ↑ is) creates a new line of normal paragraph text:
```html
<h1>Heading</h1>
                ↑
```
```html
<p>Normal paragraph text.</p>
                             ↑
```
Becomes this with the cursor (↑) positioned:
```html
<h1>Heading</h1>
<p></p>
↑
```
```html
<p>Normal paragraph text.</p>
<p></p>
↑
```

**Comment by d-e-v-esh** [comment]
Yes, that is exactly how it needs to work. I can make a sample work but making a PR would not make sense until we fix this #270 issue.

**Comment by d-e-v-esh** [comment]
@irmerk Can you please check the updates that I made in the description of #270. Any feedback would be really appreciated.

**Comment by d-e-v-esh** [comment]
@irmerk I have fixed this issue locally. I can make a pull request after the #270 is resolved and merged.
![Qt3NU2DABz](https://user-images.githubusercontent.com/59534570/117535360-0df59e80-b013-11eb-912b-9d0bcc4d6764.gif)

**Comment by dhruvinjs** [comment]
If this issue is still open can i work on this?
Can you guide me through it!

**Comment by DianaLease** [comment]
Hi @dhruvinjs , this repo isn't actively maintained at the moment so we are not encouraging new changes to it.
